src/main/scala/nl.knaw.dans.easy.deposit/StateManager.scala

```diff
@@ -82,36 +93,49 @@ case class StateManager(depositDir: File, submitBase: File) {
     }
   }
 
-  /** @return the value of bag-store.bag-id created by changeState when set to SUBMITTED */
+  /** @return the value of bag-store.bag-id which is
+   *          - created by changeState when changed to SUBMITTED
+   *          - destroyed by changeState when changed to DRAFT
+   *          or fails with:
+   *          PropertyException (if bag-store.bag-id is not found)
+   *          ConversionException (if setProperty was called with something else than a String)
+   *          IllegalArgumentException (if the value is not a valid UUID)
+   */
   def getSubmittedBagId: Try[UUID] = Try {
     // exception intercepted as not expected
     // by TriedActionResult.getOrRecoverWithActionResult
     UUID.fromString(draftProps.getString(bagIdKey))
   }
 
-  private def saveNewState(newStateInfo: StateInfo) = {
+  // TODO assemble with identifier.fedora
```

Comment:
Hint: when the dataset is archived, the `state.description` is this url. Perhaps we can make it so that this is also the case for `IN_REVIEW` and `FEDORA_ARCHIVED`? But that needs to happen in `easy-ingest-flow`.

Reply:
Can't currently check this with a manual test on deasy because ingest fails with
```
ERROR FAILED: unhandled problem during processing. Error: Deposit Agreement Creator Service failed: code=404; msg=DatasetId easy-dataset:17 does not exist
java.lang.RuntimeException: Deposit Agreement Creator Service failed: code=404; msg=DatasetId easy-dataset:17 does not exist
```

Reply:
Submitting ate least one payload file works around the problem.